bigbluebutton-html5/imports/api/audio/client/bridge/sip.js

```diff
@@ -1,85 +1,336 @@
+    };
   }
 
-  joinListenOnly(stunServers, turnServers, callbackFromManager) {
-    makeCall('listenOnlyToggle', true);
-    this._joinVoiceCallSIP({ isListenOnly: true }, stunServers, turnServers, callbackFromManager);
+  joinAudio({ isListenOnly, extension, inputStream }, managerCallback) {
+    return new Promise((resolve, reject) => {
+      const callExtension = extension ? `${extension}${this.userData.voiceBridge}` : this.userData.voiceBridge;
+
+      const callback = (message) => {
+        managerCallback(message).then(resolve);
+      };
+
+      this.callback = callback;
+
+      return this.doCall({ callExtension, isListenOnly, inputStream }, callback)
+                 .catch((reason) => {
+                   callback({
+                     status: this.baseCallStates.failed,
+                     error: this.baseErrorCodes.GENERIC_ERROR,
+                     bridgeError: reason,
+                   });
+                   reject(reason);
+                 });
+    });
   }
 
-  joinMicrophone(stunServers, turnServers, callbackFromManager) {
-    this._joinVoiceCallSIP({ isListenOnly: false }, stunServers, turnServers, callbackFromManager);
+  doCall(options) {
+    const {
+      isListenOnly,
+    } = options;
+
+    const {
+      userId,
+      name,
+      sessionToken,
+    } = this.user;
+
+    const callerIdName = [
+      userId,
+      'bbbID',
+      isListenOnly ? `LISTENONLY-${name}` : name,
+    ].join('-');
+
+    this.user.callerIdName = callerIdName;
+    this.callOptions = options;
+
+    return fetchStunTurnServers(sessionToken)
+                        .then(this.createUserAgent.bind(this))
+                        .then(this.inviteUserAgent.bind(this))
+                        .then(this.setupEventHandlers.bind(this));
   }
 
-  // Periodically check the status of the WebRTC call, when a call has been established attempt to
-  // hangup, retry if a call is in progress, send the leave voice conference message to BBB
-  exitAudio(isListenOnly, afterExitCall = () => { }) {
-    // To be called when the hangup is confirmed
-    const hangupCallback = function () {
-      console.log(`Exited Voice Conference, listenOnly=${isListenOnly}`);
+  transferCall(onTransferSuccess) {
+    return new Promise((resolve, reject) => {
+      let trackerControl = null;
 
-      // notify BBB-apps we are leaving the call if we are in listen only mode
-      if (isListenOnly) {
-        makeCall('listenOnlyToggle', false);
-      }
+      const timeout = setTimeout(() => {
+        clearTimeout(timeout);
+        trackerControl.stop();
+        this.callback({
+          status: this.baseCallStates.failed,
+          error: this.baseErrorCodes.REQUEST_TIMEOUT,
+          bridgeError: 'Timeout on call transfer' });
+        reject(this.baseErrorCodes.REQUEST_TIMEOUT);
+      }, CALL_TRANSFER_TIMEOUT);
+
+      // This is is the call transfer code ask @chadpilkey
+      this.currentSession.dtmf(1);
```

Comment:
since when we started having every single thing in the config files?

Reply:
¯\_(ツ)_/¯

Reply:
I mean, where do we draw the line of what to have and what to not have in the config files?

Reply:
@capilkey since you are referenced here - does this number look good for a config, or it will never change?

Reply:
I see configs as properties that people might want to change without rebuilding the code. That DTMF code isn't changing and if someone did change it the transfer would break. You could have it as a local constant if you don't want a number, but putting it in a config is overkill.